FreeRTOS-Plus/Source/FreeRTOS-Plus-TCP/FreeRTOS_DNS.c

```diff
@@ -598,25 +841,97 @@ TickType_t uxIdentifier = 0U;
 		and return. */
 		#if( ipconfigINCLUDE_FULL_INET_ADDR == 1 )
 		{
-			ulIPAddress = FreeRTOS_inet_addr( pcHostName );
+			/* Check if the hostname given is actually an IP-addresss. */
+			#if( ipconfigUSE_IPv6 != 0 )
+			if( xFamily == FREERTOS_AF_INET6 )
+			{
+			IPv6_Address_t xAddress_IPv6;
+			BaseType_t xResult;
+
+				/* ulIPAddress does not represent an IPv4 address here. It becomes non-zero when the look-up succeeds. */
+				xResult = FreeRTOS_inet_pton6( pcHostName, xAddress_IPv6.ucBytes );
+				if( xResult == 1 )
+				{
+					ulIPAddress = 1U;
+					if( ppxAddressInfo != NULL )
+					{
+						*( ppxAddressInfo ) = pxNew_AddrInfo( pcHostName, FREERTOS_AF_INET6, xAddress_IPv6.ucBytes );
+					}
+				}
+			}
+			else
+			#endif
+			{
+				ulIPAddress = FreeRTOS_inet_addr( pcHostName );
+				if( ( ulIPAddress != 0UL) && ( ppxAddressInfo != NULL ) )
+				{
+				uint8_t *ucBytes = ( uint8_t * ) &( ulIPAddress );
+
+					*( ppxAddressInfo ) = pxNew_AddrInfo( pcHostName, FREERTOS_AF_INET4, ucBytes );
+				}
+			}
 		}
-		#endif /* ipconfigINCLUDE_FULL_INET_ADDR == 1 */
+		#endif
 
 		/* If a DNS cache is used then check the cache before issuing another DNS
 		request. */
 		#if( ipconfigUSE_DNS_CACHE == 1 )
 		{
-			if( ulIPAddress == 0UL )
+			#if( ipconfigUSE_IPv6 != 0 )
+			if( xFamily == FREERTOS_AF_INET6 )
 			{
-				ulIPAddress = FreeRTOS_dnslookup( pcHostName );
-
-				if( ulIPAddress != 0UL )
+				if( ulIPAddress == 0UL )
 				{
-					FreeRTOS_debug_printf( ( "FreeRTOS_gethostbyname: found '%s' in cache: %lxip\n", pcHostName, ulIPAddress ) );
+				IPv46_Address_t xIPv46_Address;
+				BaseType_t xFound;
+
+					xIPv46_Address.xIs_IPv6 = pdTRUE;
+					xFound = prvProcessDNSCache( pcHostName, &( xIPv46_Address ), 0, pdTRUE, ppxAddressInfo );
+					if( xFound != 0U )
+					{
+						if( ( ppxAddressInfo != NULL ) && ( *( ppxAddressInfo ) != NULL ) )
+						{
+							struct freertos_sockaddr6 *sockaddr6 = ( struct freertos_sockaddr6 * ) ( *( ppxAddressInfo ) )->ai_addr;
+
+/*logging*/					FreeRTOS_printf( ( "prvPrepareLookup: found '%s' in cache: %pip\n",
+								pcHostName, sockaddr6->sin_addrv6.ucBytes ) );
+							ulIPAddress = 1uL;
+						}
+					}
+					else
+					{
+						/* prvGetHostByName will be called to start a DNS lookup. */
+					}
 				}
-				else
+			}
+			else
+			#endif
```

Comment:
Can we add a comment here? like so `#endif  /*   ( ipconfigUSE_IPv6 != 0 )   */`?